src/TypeScript/EslintBridgeClient/EslintBridgeStartUp.cs

```diff
@@ -19,17 +19,183 @@
+            }
+
+            return startTask.Task;
+        }
+
+        public void Dispose()
+        {
+            Stop();
+        }
+
+        private void StartServer()
+        {
+            var nodePath = FindNodePath();
+            var serverStartupScriptLocation = FindServerScriptFile();
+
+            var psi = new ProcessStartInfo
+            {
+                FileName = nodePath,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false, // required if we want to capture the error output
+                ErrorDialog = false,
+                CreateNoWindow = true,
+                Arguments = serverStartupScriptLocation
+            };
+
+            psi.EnvironmentVariables.Add("NODE_PATH", @"%APPDATA%\npm\node_modules");
+
+            Process = new Process {StartInfo = psi};
+            Process.ErrorDataReceived += OnErrorDataReceived;
+            Process.OutputDataReceived += OnOutputDataReceived;
+
+            Process.Start();
+            logger.WriteLine(Resources.INFO_ServerProcessId, Process.Id);
+            logger.LogDebug(Resources.INFO_ServerHasExisted, Process.HasExited);
+
+            Process.BeginErrorReadLine();
+            Process.BeginOutputReadLine();
+        }
+
+        private void OnOutputDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            if (string.IsNullOrEmpty(e.Data))
+            {
+                return;
+            }
+
+            logger.LogDebug("ESLINT-BRIDGE OUTPUT: " + e.Data);
+
+            var portMessage = Regex.Matches(e.Data, @"port\s+(\d+)");
+
+            if (portMessage.Count > 0)
+            {
+                var portNumber = int.Parse(portMessage[0].Groups[1].Value);
+
+                if (portNumber != 0)
+                {
+                    logger.WriteLine(Resources.INFO_ServerStarted, portNumber);
+                    startTask.SetResult(portNumber);
+                }
+            }
+        }
+
+        private void OnErrorDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            if (string.IsNullOrEmpty(e.Data))
+            {
+                return;
+            }
+
+            logger.LogDebug("ESLINT-BRIDGE ERROR: " + e.Data);
+        }
+
+        private void Stop()
+        {
+            if (Process == null || Process.HasExited)
+            {
+                logger.LogDebug("Node process has already terminated");
+                Process?.Dispose();
+                Process = null;
+                return;
+            }
+
+            Process.Kill();
+            Process?.Dispose();
+            logger.LogDebug("Node process killed");
+            Process = null;
+        }
+
+        private string FindNodePath()
+        {
+            var nodePath = nodeLocator.Locate();
+
+            if (string.IsNullOrEmpty(nodePath))
+            {
+                throw new FileNotFoundException("Could not find node.exe");
```

Comment:
Not sure that FileNotFoundException is the best one, maybe a custom exception would be better. This is so the task will fail and the caller could retry.

Reply:
Leaving aside the type of the exception for now, what overall UX do we want if node can't be found?

Should we should this message only once then stop trying to analyse JS files, or should we keep triggering analyses and display this message multiple times?

It's not directly the responsibility of this class, but it would affect how this class and the http wrapper are created and called.

Reply:
Good question. Originally I thought that it's better to retry, so the user could install node in the background and won't have to restart VS. However, if you're a mixed C# and JS dev, and you don't care about JS, you're gonna get spammed with this message each and every time you open/save a JS file. On the other hand, if it's shown only once, you might miss it in the sea of logs... So I'm really not sure. Perhaps we can start with retries and see if anyone complains? WDYT?

Reply:
As it is, the user would get errors every time a JS file is opened or saved (with the full call stack, or has that changed)? I'd say that was too noisy.

Reply:
I agree we should have a retry process to handle cases where the process has crashed/been terminated; we should just handle that behind the scenes.
I don't think we need a retry process for cases where node isn't installed. It's not unreasonable for the user to have to restart VS after they have installed node.